test/visual_regression/services/visual_testing/take_percy_snapshot.js

```diff
@@ -20,18 +20,44 @@
 import { readFileSync } from 'fs';
 import { agentJsFilename } from '@percy/agent/dist/utils/sdk-utils';
 
-export function takePercySnapshot() {
+export function takePercySnapshot(show, hide) {
   if (!window.PercyAgent) {
     return false;
   }
 
+  const queryAll = selector => [
+    ...document.querySelectorAll(selector)
+  ];
+
+  const setSelectorClass = (selectors, className, remove) => {
+    for (const selector of selectors) {
+      for (const element of queryAll(selector)) {
+        if (remove) {
+          element.classList.remove(className);
+        } else {
+          console.log(className, element);
+
+          element.classList.add(className);
+        }
+      }
+    }
+  };
+
+  const hideByDefault = show.length > 0;
   const agent = new window.PercyAgent({
     handleAgentCommunication: false
   });
 
-  const queryAll = selector => [
-    ...document.querySelectorAll(selector)
-  ];
+  if (hideByDefault) {
+    document.body.classList.add('percyHideAllByDefault');
+  } else {
+    document.body.classList.remove('percyHideAllByDefault');
```

Comment:
Just curious, is this necessary or just CYA?

Reply:
Probably not but was in case the class was not removed in a prior snapshot. I can delete it.